wgpu/src/shader/quad.wgsl

```diff
@@ -135,7 +146,30 @@ fn solid_fs_main(
         dist
     );
 
-    return vec4<f32>(mixed_color.x, mixed_color.y, mixed_color.z, mixed_color.w * radius_alpha);
+    let quad_color = vec4<f32>(mixed_color.x, mixed_color.y, mixed_color.z, mixed_color.w * radius_alpha);
+
+    if input.shadow_color.a > 0.0 {
+        let midpoint = input.pos + (input.scale / 2.0);
+        var radius = input.border_radius.a;
+
+        if input.position.x < midpoint.x && input.position.y < midpoint.y {
+            radius = input.border_radius.x;
+        } else if input.position.x < midpoint.x && input.position.y > midpoint.y {
+            radius = input.border_radius.a;
+        } else if input.position.x > midpoint.x && input.position.y < midpoint.y {
+            radius = input.border_radius.y;
+        } else if input.position.x > midpoint.x && input.position.y > midpoint.y {
+            radius = input.border_radius.z;
+        }
```

Comment:
Does the `select_border_radius` function not work for this? There is a variable `border_radius` that should have the proper value already.

Reply:
Oh yes you're right, not sure how I missed that 😄

aee28f72dcb36602a8f8792f4bf41426b93318ff